src/runtime/cuda/cuda_device_api.cc

```diff
@@ -73,9 +74,9 @@ class CUDADeviceAPI final : public DeviceAPI {
         return;
       }
       case kDeviceName: {
-        cudaDeviceProp props;
-        CUDA_CALL(cudaGetDeviceProperties(&props, ctx.device_id));
-        *rv = std::string(props.name);
+        std::string name(sizeof(cudaDeviceProp::name), 0);
```

Comment:
consider directly use 256, note that this might cause a problem if cuda driver changes its ABI and change name to a pointer.